LoopKitUI/Views/Settings Editors/TherapySettingsView.swift

```diff
@@ -337,45 +363,55 @@ struct SectionWithEdit<Content>: View where Content: View {
+        preMealTargetRange: DoubleRange(88...99),
+        workoutTargetRange: DoubleRange(99...111),
+        maximumBasalRatePerHour: 55,
+        maximumBolus: 4,
+        suspendThreshold: GlucoseThreshold.init(unit: .milligramsPerDeciliter, value: 60),
+        insulinSensitivitySchedule: InsulinSensitivitySchedule(unit: HKUnit.milligramsPerDeciliter.unitDivided(by: HKUnit.internationalUnit()), dailyItems: []),
+        carbRatioSchedule: nil,
+        basalRateSchedule: BasalRateSchedule(dailyItems: [RepeatingScheduleValue(startTime: 0, value: 0.2), RepeatingScheduleValue(startTime: 1800, value: 0.75)]))
+
+    static let preview_supportedBasalRates = [0.2, 0.5, 0.75, 1.0]
+    static let preview_supportedBolusVolumes = [5.0, 10.0, 15.0]
+
+    static let preview_viewModel = TherapySettingsViewModel(therapySettings: preview_therapySettings, supportedBasalRates: preview_supportedBasalRates, supportedBolusVolumes: preview_supportedBolusVolumes)
 
-public struct TherapySettingsView_Previews: PreviewProvider {
     public static var previews: some View {
         Group {
-            TherapySettingsView(mode: .modal,viewModel: TherapySettingsViewModel(therapySettings: preview_therapySettings, supportedBasalRates: preview_supportedBasalRates))
+            TherapySettingsView(mode: .modal, viewModel: preview_viewModel)
                 .colorScheme(.light)
                 .previewDevice(PreviewDevice(rawValue: "iPhone SE 2"))
                 .previewDisplayName("SE light (onboarding)")
-            TherapySettingsView(mode: .flow, viewModel: TherapySettingsViewModel(therapySettings: preview_therapySettings, supportedBasalRates: preview_supportedBasalRates))
+            TherapySettingsView(mode: .flow, viewModel: preview_viewModel)
                 .colorScheme(.light)
                 .previewDevice(PreviewDevice(rawValue: "iPhone SE 2"))
                 .previewDisplayName("SE light (settings)")
-            TherapySettingsView(mode: .modal, viewModel: TherapySettingsViewModel(therapySettings: preview_therapySettings, supportedBasalRates: preview_supportedBasalRates))
+            TherapySettingsView(mode: .modal, viewModel: preview_viewModel)
                 .colorScheme(.dark)
                 .previewDevice(PreviewDevice(rawValue: "iPhone XS Max"))
                 .previewDisplayName("XS Max dark (settings)")
-            TherapySettingsView(mode: .modal, viewModel: TherapySettingsViewModel(therapySettings: TherapySettings(), supportedBasalRates: nil))
+            TherapySettingsView(mode: .modal, viewModel: TherapySettingsViewModel(therapySettings: TherapySettings(), supportedBasalRates: nil, supportedBolusVolumes: nil))
                 .colorScheme(.light)
                 .previewDevice(PreviewDevice(rawValue: "iPhone SE 2"))
                 .previewDisplayName("SE light (Empty TherapySettings)")
         }
     }
 }
+
+extension HKQuantity {
+    convenience init?(unit: HKUnit, doubleValue: Double?) {
```

Comment:
I'd advice against an extension whose purpose is only to fail if one of its arguments is `nil`.
This operation is the purpose of `map` as defined on Optional:

```swift
let doubleValue: Double? = // ...
let quantity = doubleValue.map { HKQuantity(unit: unit, doubleValue: $0) } 
```

Reply:
Not fail...return `nil` if the passed-in wrapped value is `nil`.

Reply:
That's what I mean; if a function requires its arguments to be non-nil in order to produce a result, then their types should be non-optional in the signature, i.e. requiring the optionality to be handled before the function is called.

`map` is the appropriate tool for propagating optionality if the transformation is to be applied before the optionality is handled.

Reply:
In case it's helpful, Pete and I had the same conversation a little over a year ago, here: https://github.com/tidepool-org/Loop/pull/10#discussion_r273312293

Reply:
Right, ok, we're in sync.  I'm fine with `.map`, although I still think it is strange syntax for what it is doing...but I don't feel that strongly.